src/types.cpp

```diff
@@ -722,21 +723,30 @@ namespace Exiv2 {
 }                                       // namespace Exiv2
 
 #ifdef EXV_ENABLE_NLS
+
+namespace
+{
+    bool exvGettextInitialized = false;
+    std::mutex exvGettextInitializedMutex;
+}  // namespace
+
```

Comment:
I'd probably prefer static over the anonymous namespace.

Reply:
But an anonymous namespace is modern C++ and modern means better! `</sarcasm>`.

Reply:
In few places where I have worked we took the decision of preferring anonymous namespaces over `static`. Some arguments can be found here:
https://stackoverflow.com/questions/4977252/why-an-unnamed-namespace-is-a-superior-alternative-to-static
https://stackoverflow.com/questions/4422507/superiority-of-unnamed-namespace-over-static

Reply:
Please don't put a mutex (or any kind of lock) into the library.  Try to put it in the application code (exiv2.cpp).

If you feel you must put it in the library, you should enable the user to define two entry points for "lock" and "unlock".  Something like:
```
void Exiv2::SetLockCallbacks(myUnlock, myLock);
```

The library can call the user's "myLock()" and "myUnlock()" when necessary.

Having worked on embedded systems, locks are a nightmare and should be avoided.  Locks should be at the "process level".  A hidden lock in a library can dead-lock the calling process.